app/models/movie.rb

```diff
@@ -1,8 +1,11 @@
 # frozen_string_literal: true
 
 class Movie < ApplicationRecord
-  attribute :duration, :string
-  has_many :screenings
+  has_many :screenings, dependent: :destroy
   has_many :halls, through: :screenings
-  validates :title, :duration, presence: true
+  validates_associated :screenings, :halls
+  validates :title, :description, :duration, presence: true
+  validates :title, length: { maximum: 255, too_long: '%<count>s characters is the maximum allowed' }
+  validates :description, length: { maximum: 1000, too_long: '%<count>s characters is the maximum allowed' }
+  validates :duration, numericality: { only_integer: true, greater_than: 0 }
```

Comment:
here again you have validation on database level - `duration` is `integer` type. Just FYI :)
```ruby
Movie.last.update(duration: 10.7)
Movie.last.duration # 10
```
(ofc when you disable this validation)